src/map/utils/charutils.cpp

```diff
@@ -3075,30 +3075,37 @@ namespace charutils
             baseexp = GetRealExp(maxlevel, PMob->GetMLevel());
             if (baseexp != 0)
             {
-                for (uint8 i = 0; i < PChar->PParty->members.size(); ++i)
-                {
-                    exp = 0;
-                    monsterbonus = 1.0f;
-                    chainactive = false;
-                    CCharEntity* PMember = (CCharEntity*)PChar->PParty->members[i];
-                    if (PMember->getZone() == PMob->getZone())
-                    {
-                        if (map_config.exp_party_gap_penalties == 1)
+                uint8 numAllianceParties = 1;
+                if (PChar->PParty->m_PAlliance) numAllianceParties = PChar->PParty->m_PAlliance->partyList.size();

```

Comment:
is it possible to use PChar->ForAlliance and use a lambda?  I want to hide the partylist/memberlist eventually so if you could get it to use that instead (if possible) I'd like that